src/archive_converter.py

```diff
@@ -21,55 +21,44 @@
+    def __init__(self, message_dao: MessageDao) -> None:
         self._message_dao = message_dao
 
-    def update(self, data_dir: str) -> List[Message]:
+    def update(self, data_dir: str) -> Dict[str, Message]:
         """ Updates index with messages in the passed in directory.
-        Returns a list of new messages """
+        Returns a dictionary mapping new messages' ids to their corresponding message."""
 
-        new_messages : List[Message] = []
+        new_messages : Dict[str, Message] = {}
 
         for filename in os.listdir(data_dir):
             if not filename.endswith(".txt"):
                 continue
             email = generate_email_from_file(os.path.join(data_dir, filename))
             if email and not self._message_dao.get(email.id):
-                new_messages.append(email)
+                new_messages[email.id] = email
         self._add_messages_to_index(new_messages)
         return new_messages
 
-    def size(self) -> int:
-        return self._message_dao.size()
-
-    def find(self, message_id : str) -> Message:
-        message = self._message_dao.get(message_id)
-        if message is None:
-            raise ValueError(f'Could not find message: {message_id}')
-        return message
-
-    def _add_messages_to_index(self, new_messages : List[Message]) -> None:
+    def _add_messages_to_index(self, new_messages: Dict[str, Message]) -> None:
         """ Iterates through all new emails and links together emails that form a thread by populating message.children. """
 
         need_parent : List[Message] = []
 
         # First iterates through all messages to distinguish between 1.) replies and 2.) the start of a thread.
-        for message in new_messages:
-            if not self._message_dao.get(message.id):
-                self._message_dao.store(message)
+        for message in new_messages.values():
 
             # If message is a reply, it needs to be associated with an existing thread
             if message.in_reply_to is not None:
-                need_parent.append(message)
-                continue
-
-        for message in need_parent:
-            parent = self._message_dao.get(message.in_reply_to)
-            if not parent:
-                logging.info('Could not find parent email, dropping %s', message.debug_info())
-                continue
-            parent.children.append(message)
-            self._message_dao.store(parent)
+                if message.in_reply_to in new_messages:
+                    parent = new_messages.get(message.in_reply_to)
+                    parent.children.append(message)
+                else:
+                    parent = self._message_dao.get(message.in_reply_to)
+                    if not parent:
+                        logging.info('Could not find parent email, dropping %s', message.debug_info())
+                        continue
+                    parent.children.append(message)
+                    new_messages[parent.id] = parent
```

Comment:
presume we're adding the parent to the local map of new messages to ensure we store the updated message in the DB. (I'll have to read through the code some more to think if there's any adverse side effects of this)

This feels like something we should call out it in the docstring.

Reply:
Included this feature in the docstring.